app/lib/scorecard/backend.dart

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:gcloud/db.dart' as db;
+import 'package:gcloud/service_scope.dart' as ss;
+import 'package:logging/logging.dart';
+import 'package:meta/meta.dart';
+
+import '../frontend/models.dart' show Package, PackageVersion;
+import '../shared/popularity_storage.dart';
+import '../shared/utils.dart';
+import '../shared/versions.dart' as versions;
+
+import 'helpers.dart';
+import 'models.dart';
+import 'scorecard_memcache.dart';
+
+export 'models.dart';
+
+final _logger = new Logger('pub.scorecard.backend');
+
+/// Sets the active scorecard backend.
+void registerScoreCardBackend(ScoreCardBackend backend) =>
+    ss.register(#_scorecard_backend, backend);
+
+/// The active job backend.
+ScoreCardBackend get scoreCardBackend =>
+    ss.lookup(#_scorecard_backend) as ScoreCardBackend;
+
+class ScoreCardBackend {
```

Comment:
Where is the documentation on score cards? Can add you add some words on the data model and the operations done here?

Should there be tests for this code and score cards?